spec/rubocop/cop/rails/safe_navigation_with_blank_spec.rb

```diff
@@ -0,0 +1,73 @@
+# frozen_string_literal: true
+
+RSpec.describe RuboCop::Cop::Rails::SafeNavigationWithBlank do
+  subject(:cop) { described_class.new }
+
+  context 'in a conditional' do
+    it 'registers an offense on a single conditional' do
+      expect_offense(<<~RUBY)
+        return unless foo&.blank?
+        ^^^^^^^^^^^^^^^^^^^^^^^^^ Avoid calling `blank?` with the safe navigation operator in conditionals.
+      RUBY
+    end
+
+    it 'registers an offense on chained conditionals' do
+      expect_offense(<<~RUBY)
+        return unless foo&.bar&.blank?
+        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Avoid calling `blank?` with the safe navigation operator in conditionals.
+      RUBY
+    end
+
+    it 'does not register an offense on `.blank?`' do
+      expect_no_offenses(<<~RUBY)
+        return if foo.blank?
+      RUBY
+    end
+
+    describe '#autocorrect' do
```

Comment:
Thanks for the updating! Can you use `expect_correction` method for testing of autocorrect?

```diff
diff --git a/spec/rubocop/cop/rails/safe_navigation_with_blank_spec.rb b/spec/rubocop/cop/rails/safe_navigation_with_blank_spec.rb
index f334e9084..ec992c87d 100644
--- a/spec/rubocop/cop/rails/safe_navigation_with_blank_spec.rb
+++ b/spec/rubocop/cop/rails/safe_navigation_with_blank_spec.rb
@@ -6,15 +6,23 @@ RSpec.describe RuboCop::Cop::Rails::SafeNavigationWithBlank do
   context 'in a conditional' do
     it 'registers an offense on a single conditional' do
       expect_offense(<<~RUBY)
-        return unless foo&.blank?
-        ^^^^^^^^^^^^^^^^^^^^^^^^^ Avoid calling `blank?` with the safe navigation operator in conditionals.
+        do_something if foo&.blank?
+        ^^^^^^^^^^^^^^^^^^^^^^^^^^^ Avoid calling `blank?` with the safe navigation operator in conditionals.
+      RUBY
+
+      expect_correction(<<~RUBY)
+        do_something if foo.blank?
       RUBY
     end

     it 'registers an offense on chained conditionals' do
       expect_offense(<<~RUBY)
-        return unless foo&.bar&.blank?
-        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Avoid calling `blank?` with the safe navigation operator in conditionals.
+        do_something if foo&.bar&.blank?
+        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Avoid calling `blank?` with the safe navigation operator in conditionals.
+      RUBY
+
+      expect_correction(<<~RUBY)
+        do_something if foo&.bar.blank?
       RUBY
     end

@@ -23,44 +31,6 @@ RSpec.describe RuboCop::Cop::Rails::SafeNavigationWithBlank do
         return if foo.blank?
       RUBY
     end
-
-    describe '#autocorrect' do
-      context 'single conditional' do
-        let(:source) do
-          <<~RUBY
-            do_something if foo&.blank?
-          RUBY
-        end
-
-        let(:corrected_source) do
-          <<~RUBY
-            do_something if foo.blank?
-          RUBY
-        end
-
-        it 'autocorrects' do
-          expect(autocorrect_source(source)).to eq(corrected_source)
-        end
-      end
-
-      context 'chained conditionals' do
-        let(:source) do
-          <<~RUBY
-            do_something if foo&.bar&.blank?
-          RUBY
-        end
-
-        let(:corrected_source) do
-          <<~RUBY
-            do_something if foo&.bar.blank?
-          RUBY
-        end
-
-        it 'autocorrects' do
-          expect(autocorrect_source(source)).to eq(corrected_source)
-        end
-      end
-    end
   end

   context 'outside a conditional' do
```

Reply:
And can you squash your commits into one?

Reply:
@koic `expect_correction` is neat, thanks! Updated & squashed commits.